src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -253,13 +271,18 @@ def snowflake_plan(self):
                 expr_to_alias=self.expr_to_alias,
                 df_aliased_col_name_to_real_col_name=self.df_aliased_col_name_to_real_col_name,
                 source_plan=self,
+                placeholder_query=self.placeholder_query,
             )
             # set api_calls to self._snowflake_plan outside of the above constructor
             # because the constructor copy api_calls.
             # We want Selectable and SnowflakePlan to share the same api_calls.
             self._snowflake_plan.api_calls = self.api_calls
         return self._snowflake_plan
 
+    @property
+    def source_plan(self) -> Optional[LogicalPlan]:
+        return self
```

Comment:
```suggestion
        return self.snowflake_plan.source_plan
```
I think this should be the right assignment here becuase it is possible to initialize a `Selectable` with a non-None `_snowflake_plan`

Reply:
yea good idea

Reply:
done